File: web/js/service/js-test-service.js
var jsTestService = {
    hasArgs: function (v, n, resolve, reject) {
        console.log("hasArgs(" + v + ")");
        resolve({p: v.q+n, q: v.q});
    },
    noArgs: function (resolve, reject) {
        console.log("noArgs()");
        resolve();
    },
    rejectsPromise: function (resolve, reject) {
        reject(new Error('reject js error'));
    },
    throwsError: function (resolve, reject) {
        throw new Error('thrown js error');
    }
};

module.exports = {
    getService: function () {
        return jsTestService;
    }
};

File: core-services/src/main/kotlin/com/vfpowertech/jsbridge/core/services/js/JSTestService.kt
package com.vfpowertech.jsbridge.core.services.js

import com.vfpowertech.jsbridge.processor.JSGenerate
import com.vfpowertech.jsbridge.processor.JSServiceName
import nl.komponents.kovenant.Promise

@JSServiceName("jsTestService")
@JSGenerate
interface JSTestService {
    fun hasArgs(v: V, n: Int): Promise<R, Exception>
    fun noArgs(): Promise<Unit, Exception>
    fun rejectsPromise(): Promise<Unit, Exception>
    fun throwsError(): Promise<Unit, Exception>
    fun missingJSMethod(): Promise<Unit, Exception>
}